aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/AwsMetricAttributeGenerator.java

```diff
@@ -0,0 +1,222 @@
+   *
+   * <p>After this, we look for the following low-cardinality span attributes that can be used to
+   * determine the remote metric attributes:
+   *
+   * <ul>
+   *   <li>RPC
+   *   <li>DB
+   *   <li>FAAS
+   *   <li>Messaging
+   *   <li>GraphQL - Special case, if {@link SemanticAttributes#GRAPHQL_OPERATION_TYPE} is present,
+   *       we use it for RemoteOperation and set RemoteService to {@link #GRAPHQL}.
+   * </ul>
+   *
+   * <p>In each case, these span attributes were selected from the OpenTelemetry trace semantic
+   * convention specifications as they adhere to the three following criteria:
+   *
+   * <ul>
+   *   <li>Attributes are meaningfully indicative of remote service/operation names.
+   *   <li>Attributes are defined in the specification to be low cardinality, usually with a low-
+   *       cardinality list of values.
+   *   <li>Attributes are confirmed to have low-cardinality values, based on code analysis.
+   * </ul>
+   *
+   * TODO: This specific logic may change in future. Specifically, we are still deciding which HTTP
+   * and RPC attributes to use here, but this is a sufficient starting point.
+   */
+  private static void setRemoteServiceAndOperation(SpanData span, AttributesBuilder builder) {
+    if (isKeyPresent(span, AWS_REMOTE_SERVICE) || isKeyPresent(span, AWS_REMOTE_OPERATION)) {
+      setRemoteService(span, builder, AWS_REMOTE_SERVICE);
+      setRemoteOperation(span, builder, AWS_REMOTE_OPERATION);
+    } else if (isKeyPresent(span, RPC_SERVICE) || isKeyPresent(span, RPC_METHOD)) {
+      setRemoteService(span, builder, RPC_SERVICE);
+      setRemoteOperation(span, builder, RPC_METHOD);
+    } else if (isKeyPresent(span, DB_SYSTEM) || isKeyPresent(span, DB_OPERATION)) {
+      setRemoteService(span, builder, DB_SYSTEM);
+      setRemoteOperation(span, builder, DB_OPERATION);
+    } else if (isKeyPresent(span, FAAS_INVOKED_PROVIDER) || isKeyPresent(span, FAAS_INVOKED_NAME)) {
+      setRemoteService(span, builder, FAAS_INVOKED_PROVIDER);
+      setRemoteOperation(span, builder, FAAS_INVOKED_NAME);
+    } else if (isKeyPresent(span, MESSAGING_SYSTEM) || isKeyPresent(span, MESSAGING_OPERATION)) {
+      setRemoteService(span, builder, MESSAGING_SYSTEM);
+      setRemoteOperation(span, builder, MESSAGING_OPERATION);
+    } else if (isKeyPresent(span, GRAPHQL_OPERATION_TYPE)) {
+      builder.put(AWS_REMOTE_SERVICE, GRAPHQL);
+      setRemoteOperation(span, builder, GRAPHQL_OPERATION_TYPE);
+    } else {
+      logUnknownAttribute(AWS_REMOTE_SERVICE, span);
+      builder.put(AWS_REMOTE_SERVICE, UNKNOWN_REMOTE_SERVICE);
+      logUnknownAttribute(AWS_REMOTE_OPERATION, span);
+      builder.put(AWS_REMOTE_OPERATION, UNKNOWN_REMOTE_OPERATION);
+    }
+
+    // Peer service takes priority as RemoteService over everything but AWS Remote.
+    if (isKeyPresent(span, PEER_SERVICE) && !isKeyPresent(span, AWS_REMOTE_SERVICE)) {
+      setRemoteService(span, builder, PEER_SERVICE);
```

Comment:
is it expected that we have added the log entry by `logUnknownAttribute(AWS_REMOTE_SERVICE, span);` but  the `RemoteService` is actually set properly here?

Reply:
Yea I see what you mean, it's a little bit odd. I think for now it's ok. We plan to revise a bit of this logic in future, and it's not totally unreasonable for the time being. We don't expect `PEER_SERVICE` to be frequently applied. Will make a note for the future.